src/evo/deterministicmns.cpp

```diff
@@ -1541,6 +1557,18 @@ bool CheckProUpRegTx(CDeterministicMNManager& dmnman, const CTransaction& tx, gs
         return state.Invalid(TxValidationResult::TX_CONSENSUS, "bad-protx-hash");
     }
 
+    const bool is_v23_active{DeploymentActiveAfter(pindexPrev, Params().GetConsensus(), Consensus::DEPLOYMENT_V23)};
+
+    // Don't allow legacy scheme versioned transactions after upgrading to basic scheme
+    if (is_v23_active && dmn->pdmnState->nVersion >= ProTxVersion::BasicBLS && opt_ptx->nVersion == ProTxVersion::LegacyBLS) {
+        return state.Invalid(TxValidationResult::TX_CONSENSUS, "bad-protx-version-downgrade");
+    }
```

Comment:
It also shouldn't be possible to downgrade from `ExtAddr` here (`ProUpServTx` should be the only path for this)

Reply:
I think allowing downgrades from `ExtAddr` might not be the worst idea, especially since it's a new addition. The hardening of `BasicBLS` in comparison is justified since it's been deployed for a few major versions, IMO `ExtAddr` should be hardened after at least one major release since its deployment.

Reply:
Shouldn't be possible as `ProUpRegTx` is capped at `BasicBLS` and we already prevent problematic downgrade scenarios in the masternode state ([comment](https://github.com/dashpay/dash/pull/6665#discussion_r2151976729)) but these code changes have now been spun off into [dash#6729](https://github.com/dashpay/dash/pull/6729).

Reply:
I don't know... I think I'd prefer having explicit rules that simply would not allow invalid txes rather than having some relaxing logic which ignores invalid values later down the road even after v23 activation. Thoughts @PastaPastaPasta ?

Reply:
An explicit rule has been added in [dash#6729](https://github.com/dashpay/dash/pull/6729) (92984c94646fa425d69e793959ceae2ff729881a)